lib/model.js

```diff
@@ -540,7 +540,11 @@ const prepareRequest = (model) => {
       authMechanism: model.driverAuthMechanism
     });
   } else if (model.authStrategy === 'X509') {
-    req.auth = 'AUTH_TOKEN';
+    if (model.x509Username) {
+      // Username is not required with x509.
+      // Since MongoDB 3.4 it's pulled from the client certificate.
+      req.auth = 'AUTH_TOKEN';
```

Comment:
Just curious what is 'AUTH_TOKEN' here?

Reply:
We use `AUTH_TOKEN` as a flag to know where to replace the auth on the url. Since we build two urls, the safe one with `****` and the one with the actual credentials, we currently use this token lets us know where to set the auth in the url. I think we could probably skip this intermediate tokening step, but that's a bit out of the scope of this pr.